napari/_qt/widgets/qt_dims_sorter.py

```diff
@@ -46,55 +36,99 @@ def _array_in_range(arr: np.ndarray, low: int, high: int) -> bool:
     return (arr >= low) & (arr < high)
 
 
-def move_indices(axes_list: SelectableEventedList, order: Tuple[int, ...]):
-    with axes_list.events.blocker_all():
-        if tuple(axes_list) == tuple(order):
+# TODO: This function is in the current usecase of QtDimsSorter unnecessary and could be removed
+def move_indices(axis_list: SelectableEventedList, order: Tuple[int, ...]):
+    with axis_list.events.blocker_all():
+        if tuple(axis_list) == tuple(order):
             return
-        axes = [a.axis for a in axes_list]
+        axes = [a.axis for a in axis_list]
         ax_to_existing_position = {a: ix for ix, a in enumerate(axes)}
         move_list = np.asarray(
             [(ax_to_existing_position[order[i]], i) for i in range(len(order))]
         )
         for src, dst in move_list:
-            axes_list.move(src, dst)
+            axis_list.move(src, dst)
             move_list[_array_in_range(move_list[:, 0], dst, src)] += 1
         # remove the elements from the back if order has changed length
-        while len(axes_list) > len(order):
-            axes_list.pop()
+        while len(axis_list) > len(order):
+            axis_list.pop()
 
 
 class QtDimsSorter(QWidget):
-    """
+    """Qt widget for dimension / axis reordering and locking.
+
     Modified from:
     https://github.com/jni/zarpaint/blob/main/zarpaint/_dims_chooser.py
+
+    Parameters
+    ----------
+    viewer : napari.Viewer
+        Main napari viewer instance.
+    parent : QWidget
+        QWidget that holds this widget. A QtDimsSorter instances will
+        disconnect all callbacks upon closing of it's parent.
+
+    Attributes
+    ----------
+    dims : napari.components.Dims
+        Dimensions object of the current viewer, modeling slicing and displaying.
+    axis_list : napari._qt.containers.qt_axis_model.AxisList
+        Selectable evented list representing the viewer axes.
     """
 
-    def __init__(self, viewer: 'Viewer', parent=None) -> None:
+    def __init__(self, viewer: 'Viewer', parent: QWidget) -> None:
```

Comment:
I prefer to change here to pass `Dims` instance instead of the `Viewer` instance. It only requires `Dims`, and such change simplifies testing (and test time, as we do not construct complex model).

Reply:
I prefer that too.

Reply:
Done. Commit incoming shortly.